runner.go

```diff
@@ -91,12 +91,15 @@ func (r *Runtime) groupTaskAndFlagArgs(args []string) map[string][]string {
 	var currTaskName string
 	var currFlagsList []string
 	for _, arg := range args {
-		if currTaskName != "" && (strings.HasPrefix(arg, "-") || strings.HasPrefix(arg, "--")) {
+		isFlag := strings.HasPrefix(arg, "-") || strings.HasPrefix(arg, "--")
+		if isFlag {
 			// If we have identified a current task and we think the arg is a flag,
 			// add it to the list of flags we are storing for the current task.
 			// Notably, if the first flags are options to taskrunner, currTaskName will be ""
 			// and we will not group those flags with any tasks.
-			currFlagsList = append(currFlagsList, arg)
+			if currTaskName != "" {
```

Comment:
so to be clear, we mis-interpreting taskrunner flags (as opposed to taskrunner _task_ flags), causing this failure?

Reply:
Yup so essentially what was happening before is the condition was too strict since it was checking for the arg being both a flag _and_ no current task being defined.

So when we passed `taskrunner --config pathToConfig task1`, we would pass `[--config, pathToConfig, task1]` as the list of args into this fn

When we hit the first condition, we would see that currTaskName != "" so we would immediately go into the `else` statement which assumed that the arg was _not_ a flag